pkg/discovery/globalnet/globalnet.go

```diff
@@ -293,6 +293,9 @@ func ValidateGlobalnetConfiguration(globalnetInfo *GlobalnetInfo, netconfig Conf
 func GetGlobalNetworks(k8sClientset *kubernetes.Clientset, brokerNamespace string) (*GlobalnetInfo, *v1.ConfigMap, error) {
 	configMap, err := broker.GetGlobalnetConfigMap(k8sClientset, brokerNamespace)
 	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, nil, err
+		}
 		return nil, nil, fmt.Errorf("error reading configMap: %s", err)
```

Comment:
```suggestion
		return nil, nil, err
```

I feel we can just return err and let the caller check for the type of error.

Reply:
you can also do errors.Wrap(err, "error reading configMap") which I think will preserve all the other attributes (being able to call IsNotFound, etc).

Always better to use errors.Wrapf / errors.Wrap

Reply:
Yes, always use `errors.Wrap()`. Technically, this doesn’t preserve the original error’s _attributes_ directly, but it supports `Unwrap()`, which is [the documented way of handling wrapped errors](https://pkg.go.dev/errors) and [is supported by K8s’ `.Is...` functions](https://pkg.go.dev/k8s.io/apimachinery/pkg/api/errors).